sdk/web-pubsub/web-pubsub-client/README.md

```diff
@@ -0,0 +1,211 @@
+);
+```
+
+### Only allow specified endpoints
+
+```js
+const express = require("express");
+
+const { WebPubSubEventHandler } = require("@azure/web-pubsub-express");
+const handler = new WebPubSubEventHandler("chat", {
+  allowedEndpoints: [
+    "https://<yourAllowedService1>.webpubsub.azure.com",
+    "https://<yourAllowedService2>.webpubsub.azure.com"
+  ]
+});
+
+const app = express();
+
+app.use(handler.getMiddleware());
+
+app.listen(3000, () =>
+  console.log(`Azure WebPubSub Upstream ready at http://localhost:3000${handler.path}`)
+);
+```
+
+### Set custom event handler path
+
+```js
+const express = require("express");
+
+const { WebPubSubEventHandler } = require("@azure/web-pubsub-express");
+const handler = new WebPubSubEventHandler("chat", {
+  path: "customPath1"
+});
+
+const app = express();
+
+app.use(handler.getMiddleware());
+
+app.listen(3000, () =>
+  // Azure WebPubSub Upstream ready at http://localhost:3000/customPath1
+  console.log(`Azure WebPubSub Upstream ready at http://localhost:3000${handler.path}`)
+);
+```
+
+### Set and read connection state
+
+```js
+const express = require("express");
+
+const { WebPubSubEventHandler } = require("@azure/web-pubsub-express");
+
+const handler = new WebPubSubEventHandler("chat", {
+  handleConnect(req, res) {
+    // You can set the state for the connection, it lasts throughout the lifetime of the connection
+    res.setState("calledTime", 1);
+    res.success();
+  },
+  handleUserEvent(req, res) {
+    var calledTime = req.context.states.calledTime++;
+    console.log(calledTime);
+    // You can also set the state here
+    res.setState("calledTime", calledTime);
+    res.success();
+  }
+});
+
+const app = express();
+
+app.use(handler.getMiddleware());
+
+app.listen(3000, () =>
+  console.log(`Azure WebPubSub Upstream ready at http://localhost:3000${handler.path}`)
+);
+```
+
+## Troubleshooting
+
+### Enable logs
+
+You can set the following environment variable to get the debug logs when using this library.
+
+- Getting debug logs from the SignalR client library
+
+```bash
+export AZURE_LOG_LEVEL=verbose
+```
+
+For more detailed instructions on how to enable logs, you can look at the [@azure/logger package docs](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/core/logger).
+
+### Live Trace
+
+Use **Live Trace** from the Web PubSub service portal to view the live traffic.
+
+## Next steps
+
+Please take a look at the
+[samples][samples_ref]
+directory for detailed examples on how to use this library.
+
+## Contributing
+
+If you'd like to contribute to this library, please read the [contributing guide](https://github.com/Azure/azure-sdk-for-js/blob/main/CONTRIBUTING.md) to learn more about how to build and test the code.
+
+## Related projects
+
+- [Microsoft Azure SDK for Javascript](https://github.com/Azure/azure-sdk-for-js)
+
+[azure_sub]: https://azure.microsoft.com/free/
+[samples_ref]: https://github.com/Azure/azure-webpubsub/tree/main/samples/javascript/
```

Comment:
is the link to samples folder correct?

Reply:
I will update README in further pull requests as currently I don't have the link before merge